packages/@expressive-code/plugin-collapsible-sections/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import { AttachedPluginData, ExpressiveCodePlugin, replaceDelimitedValues } from '@expressive-code/core'
+import { Section, parseSections } from './utils'
+import { select } from 'hast-util-select'
+import { SectionText, sectionizeAst } from './ast'
+import { collapsibleSectionsStyleSettings, getCollapsibleSectionsBaseStyles } from './styles'
+
+export interface PluginCollapsibleSectionsOptions {
+	styleOverrides?: Partial<typeof collapsibleSectionsStyleSettings.defaultSettings> | undefined
+	summary?: SectionText | undefined
+}
+
+export function pluginCollapsibleSections(options: PluginCollapsibleSectionsOptions = {}): ExpressiveCodePlugin {
+	return {
+		name: 'Collapsible sections',
+		baseStyles: ({ theme, coreStyles }) => getCollapsibleSectionsBaseStyles(theme, coreStyles, options.styleOverrides || {}),
+		hooks: {
+			preprocessMetadata: ({ codeBlock }) => {
+				codeBlock.meta = replaceDelimitedValues(
+					codeBlock.meta,
+					({ fullMatch, key, value }) => {
+						// if we aren't interested in the entry, just return it as-is
+						if (key !== 'collapse') return fullMatch
+
+						// otherwise add our parsed data to the codeblock, so we can later modify the AST
+						const sections = parseSections(value)
+						const data = pluginCollapsibleSectionsData.getOrCreateFor(codeBlock)
+						data.sections = sections
+						return ''
+					},
+					{
+						valueDelimiters: ['"', "'", '{...}'],
+						keyValueSeparator: '=',
+					}
+				)
+			},
+			postprocessRenderedBlock: ({ codeBlock, renderData }) => {
+				const data = pluginCollapsibleSectionsData.getOrCreateFor(codeBlock)
+				const codeAst = select('pre > code', renderData.blockAst)
+				if (codeAst) {
+					codeAst.children = sectionizeAst({
+						lines: codeAst.children,
+						sections: data.sections,
+						text: options.summary,
+					})
+				}
+			},
```

Comment:
```suggestion
			postprocessRenderedBlock: ({ codeBlock, renderData, locale }) => {
				const data = pluginCollapsibleSectionsData.getOrCreateFor(codeBlock)
				if (!data.sections.length) return
				const codeAst = select('pre > code', renderData.blockAst)
				if (!codeAst) return
				codeAst.children = sectionizeAst({
					lines: codeAst.children,
					sections: data.sections,
					text: pluginCollapsibleSectionsTexts.get(locale).collapsedLines,
				})
			},
```

Passing the localized text to your function would work like this.

Also, I added a condition that skips the entire AST manipulation if no sections were defined by the user for a code block.